kong/runloop/handler.lua

```diff
@@ -1478,7 +1489,24 @@ return {
         end
 
         if enabled_headers[headers.VIA] then
-          header[headers.VIA] = SERVER_HEADER
+          local upstream_scheme = var.upstream_scheme
+          -- Kong does not support directives like 'nginx_location_proxy_http_version', so
+          -- skip checking them.
+          local proxy_http_version = ctx.proxy_http_version or
+                                     kong.configuration.proxy_http_version or
+                                     DEFAULT_PROXY_HTTP_VERSION
+
+          local kong_outbound_via
+          if upstream_scheme then
+            kong_outbound_via = concat({ string.upper(upstream_scheme) .. "/" .. proxy_http_version, SERVER_HEADER }, " ")
```

Comment:
```suggestion
            kong_outbound_via = proxy_http_version .. " " .. SERVER_HEADER
```

RFC says `For brevity, the protocol-name is omitted when the received protocol is HTTP.`:
https://datatracker.ietf.org/doc/html/rfc9110#name-via

Also never create tables in hotpath to concatenate strings.

Reply:
@bungle got it. I tried to make the header field complete, but considering avoiding table creation at hot code path, we can remove the `HTTP` prefix.

Reply:
`string.upper` need to be localized as well, this is one of the hottest codepath in the code.

Reply:
@dndx ACK.